python-package/README.rst

```diff
@@ -54,77 +61,95 @@ To install all dependencies needed to use ``pandas`` in LightGBM, append ``[pand
+- In case you prefer **gcc**, you need to install it (details for installation can be found in `Installation Guide <https://github.com/microsoft/LightGBM/blob/master/docs/Installation-Guide.rst#gcc-1>`__) and specify compilers by running ``export CXX=g++-7 CC=gcc-7`` (replace "7" with version of **gcc** installed on your machine) first.
 
 For **Windows** users, **Visual Studio** (or `VS Build Tools <https://visualstudio.microsoft.com/downloads/>`_) is needed.
 
+|
+
 Build Threadless Version
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. code:: sh
 
-    pip install lightgbm --config-settings=cmake.define.USE_OPENMP=OFF
+    pip install lightgbm --no-binary lightgbm --config-settings=cmake.define.USE_OPENMP=OFF
 
 All requirements, except the **OpenMP** requirement, from `Build from Sources section <#build-from-sources>`__ apply for this installation option as well.
 
 It is **strongly not recommended** to use this version of LightGBM!
 
+|
+
 Build MPI Version
 ~~~~~~~~~~~~~~~~~
 
 .. code:: sh
 
-    pip install lightgbm --config-settings=cmake.define.USE_MPI=ON
+    pip install lightgbm --no-binary lightgbm --config-settings=cmake.define.USE_MPI=ON
 
 All requirements from `Build from Sources section <#build-from-sources>`__ apply for this installation option as well.
 
 For **Windows** users, compilation with **MinGW-w64** is not supported.
 
 **MPI** libraries are needed: details for installation can be found in `Installation Guide <https://github.com/microsoft/LightGBM/blob/master/docs/Installation-Guide.rst#build-mpi-version>`__.
 
+|
+
 Build GPU Version
 ~~~~~~~~~~~~~~~~~
 
 .. code:: sh
 
-    pip install lightgbm --config-settings=cmake.define.USE_GPU=ON
+    pip install lightgbm --no-binary lightgbm --config-settings=cmake.define.USE_GPU=ON
 
 All requirements from `Build from Sources section <#build-from-sources>`__ apply for this installation option as well.
 
+For **macOS** users, the GPU version is not supported.
+
 **Boost** and **OpenCL** are needed: details for installation can be found in `Installation Guide <https://github.com/microsoft/LightGBM/blob/master/docs/Installation-Guide.rst#build-gpu-version>`__. Almost always you also need to pass ``OpenCL_INCLUDE_DIR``, ``OpenCL_LIBRARY`` options for **Linux** and ``BOOST_ROOT``, ``BOOST_LIBRARYDIR`` options for **Windows** to **CMake** via ``pip`` options, like
 
 .. code:: sh
 
-    pip install lightgbm \
-      --config-settings=cmake.define.USE_GPU=ON \
-      --config-settings=cmake.define.OpenCL_INCLUDE_DIR="/usr/local/cuda/include/" \
-      --config-settings=cmake.define.OpenCL_LIBRARY="/usr/local/cuda/lib64/libOpenCL.so"
+    pip install lightgbm --no-binary lightgbm --config-settings=cmake.define.USE_GPU=ON --config-settings=cmake.define.OpenCL_INCLUDE_DIR="/usr/local/cuda/include/" --config-settings=cmake.define.OpenCL_LIBRARY="/usr/local/cuda/lib64/libOpenCL.so"
```

Comment:
We cannot predict which shell user is using. For example, to split long line in PowerShell ` char should be used.